src/engine/SqliteLoggerTransport.class.js

```diff
@@ -11,89 +11,97 @@ const DEFAULT_MAX_FILE_SIZE = 2000000
+    this.tableName = options.tableName || LOGS_TABLE_NAME
     this.maxFileSize = options.maxSize || DEFAULT_MAX_FILE_SIZE
   }
 
   /**
    * Core logging method exposed to Winston.
    * @param {Object} payload - object with logging info
-   * @param {Function} callback - to be called when finished
    * @returns {undefined}
    */
-  async log(payload, callback) {
-    setImmediate(() => this.emit('logged', payload.level))
-
+  async log(payload) {
     // Perform the writing to the remote service
-    if (!this.database) {
-      await this.createLogsDatabase()
-    }
-    await this.addLog(payload.timestamp, payload.level, payload.source, payload.message)
+    await this.addLog(payload.time, payload.level, payload.scope, payload.source, payload.message)
 
     const logFile = fs.statSync(this.fileName)
     if (logFile.size > this.maxFileSize) {
       await this.deleteOldLogs()
     }
-
-    if (callback) callback()
-  }
-
-  /**
-   * Initiate SQLite3 database and create the logs table.
-   * @return {void}
-   */
-  async createLogsDatabase() {
-    this.database = await sqlite.open({ filename: this.fileName, driver: sqlite3.cached.Database })
-
-    const query = `CREATE TABLE IF NOT EXISTS ${LOGS_TABLE_NAME} (
-                    id INTEGER PRIMARY KEY, 
-                    timestamp DATE,
-                    level TEXT,
-                    source TEXT,
-                    message TEXT
-                  );`
-    const stmt = await this.database.prepare(query)
-    await stmt.run()
   }
 
   /**
    * Add logs
    * @param {string} timestamp - The timestamp
    * @param {string} level - The level
-   * @param {string} source - The source
+   * @param {string} scope - The scope (south/north connector, engine...)
+   * @param {string} source - The source file
    * @param {string} message - The message
    * @return {void}
    */
-  async addLog(timestamp, level, source, message) {
-    const query = `INSERT INTO ${LOGS_TABLE_NAME} (timestamp, level, source, message) 
+  async addLog(timestamp, level, scope, source, message) {
+    const query = `INSERT INTO ${this.tableName} (timestamp, level, scope, source, message)
                    VALUES (?, ?, ?, ?)`
     const stmt = await this.database.prepare(query)
-    await stmt.run(timestamp, level, source, message)
+    await stmt.run(timestamp, level, scope, source, message)
   }
 
   /**
    * Delete old logs.
    * @return {void}
    */
   async deleteOldLogs() {
-    const query = `DELETE FROM ${LOGS_TABLE_NAME} 
-                  WHERE id IN (
-                    SELECT id FROM ${LOGS_TABLE_NAME} 
-                    ORDER BY id ASC 
+    const query = `DELETE FROM ${this.tableName}
+                    WHERE id IN (
+                    SELECT id FROM ${this.tableName}
+                    ORDER BY id ASC
                     LIMIT ?
                     )`
     const stmt = await this.database.prepare(query)
     await stmt.run(NUMBER_OF_RECORDS_TO_DELETE)
   }
```

Comment:
should we log the number of lines that were deleted for audit?

Reply:
I'm not sure we should, we could have this edge case where we remove one line of log (as it is now). That means we log a line every time we remove a line, etc...